### PR TITLE
[Migrate Simple Payments] Custom amount click opens custom amount section

### LINKED ISSUE #11202: [Migrate Simple Payments] Add custom amount click opens custom amount section

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #11202

### Description

The PR adds an opening of addition of a custom amount screen from the bottom sheet

### Testing instructions

* More -> Payments -> Collect Payment -> Add a custom Amount
* Notice that the custom amount screen is opened

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/3824d9b6-d2a8-4fcd-bf73-cfd9520886f3

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 1]
```diff
@@ -1048,11 +1050,20 @@ class OrderCreateEditFormFragment :
...
+    private fun setupHandleResults(fragmentOrderCreateEditFormBinding: FragmentOrderCreateEditFormBinding) {
         handleDialogResult<OrderStatusUpdateSource>(
             key = KEY_ORDER_STATUS_RESULT,
             entryId = R.id.orderCreationFragment
         ) { viewModel.onOrderStatusChanged(Order.Status.fromValue(it.newStatus)) }
+        handleDialogNotice(
+            key = OrderCreateEditSimplePaymentsMigrationBottomSheetFragment.KEY_ON_ADD_CUSTOM_AMOUNT_CLICKED_NOTICE,
+            entryId = R.id.orderCreationFragment
+        ) {
+            fragmentOrderCreateEditFormBinding.root.post {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: That's called before navigation component thinks that we are on this fragment, therefore this workaround. If you have any suggestion how to make this more clear would be nice

**Comment by backwardstruck** [comment]
Tested. Later we can set a maximum length for the custom amount field:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 1 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: The only code change I found that might work is something like this but I couldn't actually get it to work:

```
lifecycleScope.launch {
    lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
        navigateToCustomAmountsDialog()
    }
}
```

Maybe just a comment for clarity?

**Review by backwardstruck — APPROVED** [review]

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt]: Good point. Extracted to a method and added a comment - 514cc3ada282b65c58ec5019d33ab3901ab27d83

**Pull request merged**